flink-connector-mysql-cdc/src/test/java/com/ververica/cdc/connectors/mysql/source/MySqlSourceITCase.java

```diff
@@ -248,6 +249,22 @@ public void testTaskManagerFailoverForNewlyAddedTableWithAheadBinlog() throws Ex
                 "address_beijing");
     }
 
+    @Test
+    public void testConsumingTableWithoutPrimaryKey() {
```

Comment:
What's the diff between the ITCase and following unit test?

Reply:
Basically they are testing the same functionality on different level. Personally I prefer to keep both the unit and integration case because unit test doesn't touch the logic that the exception would finally trigger a job failover.

Reply:
make sense to me